jabgui/src/main/java/org/jabref/gui/theme/ThemeManager.java

```diff
@@ -57,114 +58,158 @@ public class ThemeManager {
         // Always returns something even if the native library is not available - see https://github.com/dukke/FXThemes/issues/15
         this.themeWindowManager = ThemeWindowManagerFactory.create();
 
         this.baseStyleSheet = StyleSheet.create(Theme.BASE_CSS).get();
         this.theme = workspacePreferences.getTheme();
         this.isDarkMode = Theme.EMBEDDED_DARK_CSS.equals(this.theme.getName());
 
-        initializeWindowThemeUpdater(this.isDarkMode);
+        initializeWindowThemeUpdater();
 
         // Watching base CSS only works in development and test scenarios, where the build system exposes the CSS as a
         // file (e.g. for Gradle run task it will be in build/resources/main/org/jabref/gui/Base.css)
         addStylesheetToWatchlist(this.baseStyleSheet, this::baseCssLiveUpdate);
         baseCssLiveUpdate();
 
-        BindingsHelper.subscribeFuture(workspacePreferences.themeProperty(), theme -> updateThemeSettings());
-        BindingsHelper.subscribeFuture(workspacePreferences.themeSyncOsProperty(), theme -> updateThemeSettings());
-        BindingsHelper.subscribeFuture(workspacePreferences.shouldOverrideDefaultFontSizeProperty(), should -> updateFontSettings());
-        BindingsHelper.subscribeFuture(workspacePreferences.mainFontSizeProperty(), size -> updateFontSettings());
-        BindingsHelper.subscribeFuture(Platform.getPreferences().colorSchemeProperty(), colorScheme -> updateThemeSettings());
+        BindingsHelper.subscribeFuture(workspacePreferences.themeProperty(), _ -> updateThemeSettings());
+        BindingsHelper.subscribeFuture(workspacePreferences.themeSyncOsProperty(), _ -> updateThemeSettings());
+        BindingsHelper.subscribeFuture(workspacePreferences.shouldOverrideDefaultFontSizeProperty(), _ -> updateFontSettings());
+        BindingsHelper.subscribeFuture(workspacePreferences.mainFontSizeProperty(), _ -> updateFontSettings());
+        BindingsHelper.subscribeFuture(Platform.getPreferences().colorSchemeProperty(), _ -> updateThemeSettings());
         updateThemeSettings();
     }
 
-    private void initializeWindowThemeUpdater(boolean darkMode) {
-        this.isDarkMode = darkMode;
+    /// Installs the base and additional css files as stylesheets in the given scene.
+    public void installCss(@NonNull Scene scene) {
+        // For unclear reason, unless this is wrapped in runLater, IndexOutOfBounds will be thrown, despite
+        // all the invocation to this method come directly from the UI thread
+        UiTaskExecutor.runInJavaFXThread(() -> {
+            List<String> stylesheets = Stream
+                    .of(baseStyleSheet.getSceneStylesheet(),
+                            theme.getAdditionalStylesheet().map(StyleSheet::getSceneStylesheet).orElse(null)
+                    ).filter(Objects::nonNull)
+                    .map(URL::toExternalForm)
+                    .toList();
+            scene.getStylesheets().setAll(stylesheets);
+        });
+    }
```

Comment:
Some race condition. I remember last time I was working with the thememanager about 2 years ago i had the same issue.